Library/Homebrew/cmd/commands.rb

```diff
@@ -8,6 +8,19 @@ module Homebrew
   module_function
 
   def commands
+    valid_options = ["--quiet",
+                     "--include-aliases"].freeze
+    invalid_options_by_user = []
+    ARGV.options_only.each do |option|
+      invalid_options_by_user << option unless valid_options.include?(option)
+    end
+    unless invalid_options_by_user.empty?
+      odie <<-EOS.undent
+        Invalid option#{"s" if invalid_options_by_user.length>1} provided: #{invalid_options_by_user.join " "}
+        #{"Valid option#{"s" if valid_options.length>1}: #{valid_options.join " "}" unless valid_options.empty?}
```

Comment:
Nervermind. Please see my latest commit. It prints errors of the following type:

```
Error: 2 Invalid Options Provided: --uq --opo
Only 2 Options Valid: --quiet --include-aliases
```

```
Error: 1 Invalid Option Provided: --uq
Only 2 Options Valid: --quiet --include-aliases
```

```
Error: 1 Invalid Option Provided: --uq
Only 1 Option Valid: --quiet
```

Let me know if you prefer something else!